packages/ckeditor5-utils/src/focustracker.ts

```diff
@@ -85,23 +162,77 @@ export default class FocusTracker extends /* #__PURE__ */ DomEmitterMixin( /* #_
 			throw new CKEditorError( 'focustracker-add-element-already-exist', this );
 		}
 
-		this.listenTo( element, 'focus', () => this._focus( element ), { useCapture: true } );
-		this.listenTo( element, 'blur', () => this._blur(), { useCapture: true } );
+		this.listenTo( element, 'focus', () => {
+			// @if CK_DEBUG_FOCUSTRACKER // console.log( `"${ getName( this ) }": Focus with useCapture on DOM element` );
+
+			const externalFocusedViewInSubtree = this.externalViews.find( view => isExternalViewSubtreeFocused( element, view ) );
+
+			if ( externalFocusedViewInSubtree ) {
+				this._focus( externalFocusedViewInSubtree.element! );
+			} else {
+				this._focus( element );
+			}
+		}, { useCapture: true } );
+
+		this.listenTo( element, 'blur', () => {
+			// @if CK_DEBUG_FOCUSTRACKER // console.log( `"${ getName( this ) }": Blur with useCapture on DOM element` );
+
+			this._blur();
+		}, { useCapture: true } );
+
 		this._elements.add( element );
 	}
 
 	/**
-	 * Stops tracking the specified element and stops listening on this element.
+	 * Removes a DOM element from the focus tracker.
 	 */
-	public remove( element: Element ): void {
+	private _removeElement( element: Element ): void {
+		if ( this._elements.has( element ) ) {
```

Comment:
I had to remove the `element` first before calling `this._blur()` because `this._blur()` now has an internal logic that analyzes `_elements` and makes decisions. 

https://github.com/ckeditor/ckeditor5/blob/c44ec5036197e5f7a2b35a0152f62b0eff0e43c0/packages/ckeditor5-utils/src/focustracker.ts#L274-L279

Preserving the old order of things caused issues so now I do `this._elements.delete( element );` first and then `this._blur()`. You can reverse this change to see some tests failing. 

```
Chrome Headless 130.0.0.0 (Mac OS 10.15.7) FocusTracker remove() for views without focus tracker should update state upon removing view#element if focused FAILED
        AssertionError: expected true to be false
            at Context.<anonymous> (/var/folders/mg/zmg3mvns5vb0zg5jlbk347qc0000gn/T/_karma_webpack_722106/commons.js:96449:44)
...
Chrome Headless 130.0.0.0 (Mac OS 10.15.7) FocusTracker remove() for views with focus tracker should update the focus tracker's state upon removing view#element (if view#element was focused) FAILED
        AssertionError: expected true to be false
            at Context.<anonymous> (/var/folders/mg/zmg3mvns5vb0zg5jlbk347qc0000gn/T/_karma_webpack_722106/commons.js:96544:44)
```